components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -5322,6 +5324,36 @@ private void saveAPIDefinition(APIProduct apiProduct, String apiDefinitionJSON,
         }
     }
 
+    @Override
+    public void addAPIProductSwagger(Map<API, List<APIProductResource>> apiToProductResourceMapping, APIProduct apiProduct)
+            throws APIManagementException {
+        APIDefinition parser = new OAS3Parser();
+        SwaggerData swaggerData = new SwaggerData(apiProduct);
+        String apiProductSwagger = parser.generateAPIDefinition(swaggerData);
+
```

Comment:
Shall we remove this empty line? Please check other places as well.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/9604/commits/08c2e6b56e8dd41ef907b3140d4601a985a2307d